### PR TITLE
overhaul of LCA.to_dataframe()

### LINKED ISSUE #31: to_dataframe sorting
Hi

I was having some trouble with `to_dataframe` and I see now it is commented. In case this method will be used at any point, I think `stacked.sort()` should be replaced by something like `stacked = stacked[:,stacked[0,:].argsort()]` (or anything more elegant).. otherwise the ranking is not well sorted.

### PR DESCRIPTION
**Pull request by BenPortner**
- should fix #31 
- use pandas methods for sorting --> syntax is easier to understand (in my opinion) and easier to extend (e.g. sorting on other columns, adding new columns)
- drawback: slightly slower than the solution proposed by @mfastudillo

### CODE REVIEW

**Comment by cmutel** [comment]
Hi @BenPortner! Sorry, I am not sure why, but I am only seeing this now.

This is a great idea; 2.5 will make it slightly more complicated, but I will merge soon, as soon as I have time to resolve the conflicts. Thanks a lot!

File: bw2calc/lca.py [hunk 1]
```diff
@@ -517,39 +517,47 @@ def redo_lcia(self, demand=None):
...
-    #                 activity.get("location"),
-    #                 stacked[3, length - x - 1],
-    #             )
-    #         )
-    #     return pandas.DataFrame(data, columns=["Activity", "Flow", "Region", "Amount"])
+    def to_dataframe(self, cutoff=200):
+        """Return all nonzero elements of characterized inventory as Pandas dataframe"""
+        # assert mapping, "This method doesn't work with independent LCAs"
+        assert (
+            pandas
```
Comment (tngTUDOR) [inline bw2calc/lca.py:524]: pandas is now part of the install_requires of the package, we could leave this out.

File: bw2calc/lca.py [hunk 2]
```diff
@@ -517,39 +517,47 @@ def redo_lcia(self, demand=None):
...
+        ), "This method requires the `pandas` (http://pandas.pydata.org/) library"
+        assert hasattr(
+            self, "characterized_inventory"
+        ), "Must do LCIA calculation first"
+
+        from bw2data import get_activity
```
Comment (tngTUDOR) [inline bw2calc/lca.py:530]: `bw2data` is an optional dependency of `bw2calc`. Please see the way this is handled in `MultiLCA` (regarding `bw2data.calculation_setups`). 
(This is the opposite of the pandas requirement above)

Comment (tngTUDOR) [reply bw2calc/lca.py:530]: For example

```python3
try:
    from bw2data import get_activity
except ImportError:
    get_activity = None
...
if get_activity is None:
            raise ImportError("`bw2data` is required for this functionality")
```

Comment (tngTUDOR) [reply bw2calc/lca.py:530]: Actually, maybe we can implement in a way that it works even without `get_activity`.
`get_activity` should be used if we have it available, and hence the returned DataFrame would have a name column, but if we don't have it, we could use whatever the `self.dicts.biosphere.reversed` provide.
I'm not sure though if these (`self.dicts.X`) are systematically available.

**Comment by tngTUDOR** [comment]
I might be wrong, but I have the impression that this PR concerns 2 different, unrelated issues: #31 and #32.
It would be easier if we had only one issue to fix per PR @BenPortner.

**Comment by cmutel** [comment]
See https://github.com/brightway-lca/brightway2-regional/blob/master/bw2regional/export.py#L29 for another idea on dataframes.

**Comment by BenPortner** [comment]
Thanks for checking the PR @tngTUDOR. I will address the raised issues once https://github.com/brightway-lca/brightway2-io/pull/114 and https://github.com/brightway-lca/brightway2-data/pull/91 are solved, as I consider these high(er) priority.

**Comment by cmutel** [comment]
Closing this in favor of 3f360c913f8d257aeabe4a7a754a1939de506c1a, which follow the conventions introduced in [bw2data](https://github.com/brightway-lca/brightway2-data/commit/586cb642c9fc65d565480daeee0a76a25f1075b7), relative and absolute cutoffs, manual specification of row and col dicts, works with and without `bw2data`, and allows for dataframes from any matrix.

**Pull request closed without merging**